sdk/samples/iot/aziot_esp32/aziot_esp32.ino

```diff
@@ -0,0 +1,277 @@
+      break;
+    case MQTT_EVENT_SUBSCRIBED:
+      Logger.Info("MQTT event MQTT_EVENT_SUBSCRIBED");
+      break;
+    case MQTT_EVENT_UNSUBSCRIBED:
+      Logger.Info("MQTT event MQTT_EVENT_UNSUBSCRIBED");
+      break;
+    case MQTT_EVENT_PUBLISHED:
+      Logger.Info("MQTT event MQTT_EVENT_PUBLISHED");
+      break;
+    case MQTT_EVENT_DATA:
+      Logger.Info("MQTT event MQTT_EVENT_DATA");
+      break;
+    case MQTT_EVENT_BEFORE_CONNECT:
+      Logger.Info("MQTT event MQTT_EVENT_BEFORE_CONNECT");
+      break;
+    default:
+      Logger.Info("MQTT event UNKNOWN");
+      break;
+  }
+}
+
+static void initializeIoTHubClient()
+{
+  if (az_result_failed(az_iot_hub_client_init(
+          &client,
+          az_span_create((uint8_t*)host, strlen(host)),
+          az_span_create((uint8_t*)device_id, strlen(device_id)),
+          NULL)))
+  {
+    Logger.Info("Failed initializing Azure IoT Hub client");
+    return;
+  }
+
+  size_t client_id_length;
+  if (az_result_failed(az_iot_hub_client_get_client_id(
+          &client, mqtt_client_id, sizeof(mqtt_client_id) - 1, &client_id_length)))
+  {
+    Logger.Info("Failed getting client id");
+    return;
+  }
+
+  // Get the MQTT user name used to connect to IoT Hub
+  if (az_result_failed(az_iot_hub_client_get_user_name(
+          &client, mqtt_username, sizeofarray(mqtt_username), NULL)))
+  {
+    Logger.Info("Failed to get MQTT clientId, return code");
+    return;
+  }
+
+  Logger.Info("Client ID: " + String(mqtt_client_id));
+  Logger.Info("Username: " + String(mqtt_username));
+}
+
+static void initializeMqttClient()
+{
+  sasToken.Generate(SAS_TOKEN_DURATION_IN_MINUTES);
+
+  esp_mqtt_client_config_t mqtt_config;
+  memset(&mqtt_config, 0, sizeof(mqtt_config));
+  mqtt_config.uri = mqtt_broker_uri;
+  mqtt_config.port = port;
+  mqtt_config.client_id = mqtt_client_id;
+  mqtt_config.username = mqtt_username;
+  mqtt_config.password = (const char*)az_span_ptr(sasToken.Get());
+  mqtt_config.keepalive = 30;
+  mqtt_config.disable_clean_session = 0;
+  mqtt_config.disable_auto_reconnect = false;
+  mqtt_config.event_handle = mqtt_event_handler;
+  mqtt_config.user_context = NULL;
+  mqtt_config.cert_pem = (const char*)ca_pem;
+
+  mqtt_client = esp_mqtt_client_init(&mqtt_config);
+
+  if (mqtt_client == NULL)
+  {
+    Logger.Error("Failed creating mqtt client");
+    return;
+  }
+
+  esp_err_t start_result = esp_mqtt_client_start(mqtt_client);
+
+  if (start_result != ESP_OK)
+  {
+    Logger.Error("Could not start mqtt client; error code:" + start_result);
+  }
+  else
+  {
+    Logger.Info("MQTT client started");
+  }
+}
+
+static uint32_t getSecondsSinceEpoch() { return (uint32_t)time(NULL); }
+
+static void establishConnection()
+{
+  connectToWiFi();
+  initializeTime();
+  initializeIoTHubClient();
+  initializeMqttClient();
+}
+
+void setup() { establishConnection(); }
+
+static void getTelemetryPayload(az_span payload, az_span* out_payload)
+{
+  *out_payload = payload;
```

Comment:
do you really want to use out_payload as a temp var?

Reply:
yeah, well. It saves a variable. Anyway, I'll create a temp var.